pkg/statistics/handle/storage/read.go

```diff
@@ -563,23 +565,30 @@ func TableStatsFromStorage(sctx sessionctx.Context, snapshot uint64, tableInfo *
 		}
 	}
 	// If DROP STATS executes, we need to reset the stats version to 0.
+	// Only reset if we actually have columns/indices in the table. If all stats were skipped
+	// due to lazy loading, we should keep the StatsVer that was set from the storage row.
 	if table.StatsVer != statistics.Version0 {
```

Comment:
I am not sure when this situation would occur. When would we see the table’s stats version being non zero while all column and index stats versions are zero? I am not sure if we actually have any cases where this condition can be triggered. I think we may need to assert that this code path is reachable after executing drop stats.

Reply:
I checked that this logic has been covered by other tests. So resolved the conversation here.